### ISSUE TITLE
Better documentation

### ISSUE THREAD

**Issue by bluebaroncanada** [phpseclib/phpseclib#1149]
Documentation is awful.

For instance, sign says it returns a string, but if it fails, it clearly returns false.  No where in the documentation that I can find does it tell me to check for false, and thus, I'm left wondering if that's the only thing that indicates failure especially when it returns calls to two different other functions.

**Comment by terrafrost** [comment]
I agree. The documentation is on github for people to make changes to if they're so inclined. But I guess people would rather make PR's to phpseclib/phpseclib than they would to phpseclib/docs. The fact that you opened this issue on phpseclib/phpseclib instead of phpseclib/docs just re-enforces this notion.

My thoughts: I could spend a month or two redoing the documentation completely while phpseclib languishes. But I'm not going to do that. When the master branch is getting close to being released as 3.0.0 I'll do a complete overhaul of the documentation to coincide with the release of the new version of 3.0.0.

My time is limited. I have a 40+h job that doesn't involve phpseclib at all and don't have the ability to do phpseclib dev full time. This means that new features I want to be released ASAP take longer than I'd like. And stopping all work on those new features so I can completely redo the documentation would delay the release of features that I already think are taking too much time to get released.

**Comment by bluebaroncanada** [comment]
Well, I know you would like me to go PR docs instead of writing you this issue, however, I know less about the project than you, and I work a 40+ hr job.  
This is probably a lesson in proper documentation as part of a well designed project.

**Comment by terrafrost** [comment]
Well it's not just you lol. phpseclib/phpseclib has had 66 contributors. phpseclib/docs has had 2 contributors.

People who are familiar enough with how phpseclib works to make PR's to phpseclib itself presumably would be familiar enough with phpseclib to update the docs bu they don't.

I mean, I'm grateful for all the PR's that phpseclib/phpseclib has gotten. The point I'm just trying to make is that, sure, I've been neglecting the documentation, but so have a lot of other people who probably have a good enough understanding of phpseclib to make meaningful updates to the docs.

idk why that is but that's just the way it is none-the-less lol
